warmup/schemas/postgres/2020042101.sql

```diff
@@ -0,0 +1,63 @@
+
+CREATE TABLE IF NOT EXISTS entities_metadata (
+  _id varchar(32) NOT NULL,
+  name varchar(64) NOT NULL,
+  value text NOT NULL,
+  FOREIGN KEY (_id) REFERENCES entities (_id) ON DELETE CASCADE ON UPDATE CASCADE
+);
+
+CREATE INDEX CONCURRENTLY IF NOT EXISTS m_value ON entities_metadata (value);
+CREATE INDEX CONCURRENTLY IF NOT EXISTS m_name ON entities_metadata (name);
+CREATE INDEX CONCURRENTLY IF NOT EXISTS m__id ON entities_metadata (_id);
+
+INSERT INTO entities_metadata SELECT _id, name, value FROM metadata WHERE collection = 'entities';
+
+
+
+CREATE TABLE IF NOT EXISTS config_metadata (
+  _id varchar(32) NOT NULL,
+  name varchar(64) NOT NULL,
+  value text NOT NULL,
+  FOREIGN KEY (_id) REFERENCES config (_id) ON DELETE CASCADE ON UPDATE CASCADE
+);
+
+CREATE INDEX CONCURRENTLY IF NOT EXISTS m_value ON config_metadata (value);
+CREATE INDEX CONCURRENTLY IF NOT EXISTS m_name ON config_metadata (name);
+CREATE INDEX CONCURRENTLY IF NOT EXISTS m__id ON config_metadata (_id);
+
+INSERT INTO config_metadata SELECT _id, name, value FROM metadata WHERE collection = 'config';
+
+
+
+CREATE TABLE IF NOT EXISTS reader_metadata (
+  _id varchar(32) NOT NULL,
+  name varchar(64) NOT NULL,
+  value text NOT NULL,
+  FOREIGN KEY (_id) REFERENCES reader (_id) ON DELETE CASCADE ON UPDATE CASCADE
+);
```

Comment:
UNIQUE constraint could go here. Basically in a fairly new (April 1st) Install, I had duplicate values stored which were from the application when running the INSERT INTO SELECT FROM WHERE part of the queries

Reply:
Have to check how this affects the storage of array values.

Can't remember whether arrays are serialised in metadata or stored multiple times (I think the latter). Switched between projects so many times I can't remember off the top of my head